lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
Can't you just use something like `if (str == +str)` to distinguish numbers with some accommodation for `0`, `-0` and the other special numbers?